pkg/env/metrics.go

```diff
@@ -1,31 +1,90 @@
 package env
 
-import "net"
+import (
+	"net"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/fileutils"
+)
 
 const (
-	defaultMetricsSetting = ":9090"
+	// TLSCertFileName is the tls certificate filename.
+	TLSCertFileName = "tls.crt"
+	// TLSKeyFileName is the private key filename.
+	TLSKeyFileName = "tls.key"
+
+	defaultHTTPPort  = ":9090"
+	defaultHTTPSPort = "disabled"
 )
 
 var (
-	// MetricsSetting has the :port or host:port string for listening for metrics/debug server
-	MetricsSetting = RegisterSetting("ROX_METRICS_PORT", WithDefault(defaultMetricsSetting))
+	// MetricsPort has the :port or host:port string for listening for metrics/debug server.
+	MetricsPort = RegisterSetting("ROX_METRICS_PORT", WithDefault(defaultHTTPPort))
+	// SecureMetricsPort has the :port or host:port string for listening for metrics/debug server.
+	SecureMetricsPort = RegisterSetting("ROX_SECURE_METRICS_PORT", WithDefault(defaultHTTPSPort))
+	// SecureMetricsCertDir has the server's TLS certificate files.
+	SecureMetricsCertDir = RegisterSetting("ROX_SECURE_METRICS_CERT_DIR", WithDefault("/run/secrets/stackrox.io/monitoring-tls"))
+	// SecureMetricsClientCANamespace has the namespace that contains the client CA.
+	SecureMetricsClientCANamespace = RegisterSetting("ROX_SECURE_METRICS_CLIENT_CA_NS", WithDefault("kube-system"))
+	// SecureMetricsClientCAConfigMap has the config map that contains the client CA.
+	SecureMetricsClientCAConfigMap = RegisterSetting("ROX_SECURE_METRICS_CLIENT_CA_CFG", WithDefault("extension-apiserver-authentication"))
 )
 
-// ValidateMetricsSetting returns an error if the environment variable is invalid.
-func ValidateMetricsSetting() error {
-	val := MetricsSetting.Setting()
-	if val == "disabled" {
-		return nil
-	}
+func validatePort(setting Setting) error {
+	val := setting.Setting()
 	addr, err := net.ResolveTCPAddr("tcp", val)
 	if err != nil {
 		return err
 	}
-	log.Debugf("%s=%s, resolved to %+v", MetricsSetting.EnvVar(), val, addr)
+	log.Debugf("%s=%s, resolved to %+v", setting.EnvVar(), val, addr)
 	return nil
 }
 
-// MetricsEnabled returns true if the metrics/debug http server should be started
+func validateTLS() error {
+	certFile := filepath.Join(SecureMetricsCertDir.Setting(), TLSCertFileName)
+	if ok, err := fileutils.Exists(certFile); !ok {
+		if err != nil {
+			log.Warnf("failed to validate secure metrics certificate file %q", certFile)
```

Comment:
Why the extra log statement, AFAICT the result of `validateTLS` will be logged anyways?

Reply:
Ah the point was to also log `err`, which is only returned when there was a problem with stat'ing the file. Whereas a new error is return by `validateTLS` when no file was found, but the check itself was successful but simply returned "file does not exist".